src/cloud_service.h

```diff
@@ -51,33 +50,13 @@ enum CloudServicePublishFlags {
     FULL_ACK = 0x01 // full end-to-end acknowledgement
 };
 
-typedef std::function<int(CloudServiceStatus status, JSONValue *, const void *context)> cloud_service_cb_t;
-
-typedef std::function<int(CloudServiceStatus status, JSONValue *, const char *, const void *context)> cloud_service_send_cb_t;
-
-class cloud_service_handler_t
-{
-public:
-    CloudServicePublishFlags cloud_flags;
-    cloud_service_cb_t cb;
-    // match on this cmd (or blank for don't care)
-    char cmd[CLOUD_MAX_CMD_LEN + 1];
-    // match on this req_id (or 0 don't care)
-    uint32_t req_id;
-    // fail on match after a timeout, or 0 for never timeout
-    uint32_t timeout_ms;
-    const void *context;
-    uint32_t t0;
-    CloudServiceStatus status;
-};
+using cloud_service_ack_callback = std::function<int(CloudServiceStatus, JSONValue *, String&&)>;
```

Comment:
Is it ever expected that the ack callback will be interested in the JSON sent from the server? Or is this an artifact from having ack and command handlers share the same code? If it's not needed, some of the code can be even further simplified.

Reply:
I don't believe there is any data associated with the ack.  It is fine to assume that it is not needed.